lighthouse-core/audits/bootup-time.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit');
+const DevtoolsTimelineModel = require('../lib/traces/devtools-timeline-model');
+const Util = require('../report/v2/renderer/util.js');
+
+class BootupTime extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'bootup-time',
+      description: 'JavaScript boot-up time is high (> 4s)',
+      failureDescription: 'JavaScript boot-up time is too high.',
+      helpText: 'Consider reducing the time spent parsing, compiling and executing JS. ' +
+        'You may find delivering smaller JS payloads helps with this.',
+      requiredArtifacts: ['traces'],
+    };
+  }
+
+  /**
+   * @param {!Array<TraceEvent>=} trace
+   * @return {!Map<string, Number>}
+   */
+  static getExecutionTimingsByURL(trace) {
+    const timelineModel = new DevtoolsTimelineModel(trace);
+    const bottomUpByName = timelineModel.bottomUpGroupBy('URL');
+    const result = new Map();
+    bottomUpByName.children.forEach((value, url) => {
+      // when url is "", we skip it
+      if (!url) {
+        return;
+      }
+
+      const evaluateTime = value.children.get('EvaluateScript:@' + url) || {};
```

Comment:
we could do this but i'd rather iterate over the `perUrlNode` children (which would be `perTaskPerUrlNode` i guess)...

so iterate over them.. then grab  `perTaskPerUrlNode.event.name`. This should give you `v8.compile` and `EvaluateScript`. We can then reuse this stuff: https://github.com/GoogleChrome/lighthouse/blob/5fbff87ef9c8c18d51d05e6a07f5631d6edde8b6/lighthouse-core/audits/page-execution-timings.js#L17-L98

and we'll want any children with events of name in the 3 `script*`categories. (also note: many of those task names are for instant events so they have 0 duration. ;)

Reply:
we can exclude GC from the report for now but might as well collect it here and put in extInfo ?  or ignore it and leave a comment saying so..